Mathlib/RingTheory/MvPowerSeries/LinearTopology.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández
+-/
+import Mathlib.Data.Finsupp.Interval
+import Mathlib.RingTheory.MvPowerSeries.PiTopology
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.Nonarchimedean.Bases
+import Mathlib.RingTheory.TwoSidedIdeal.Operations
+
+/-! # Linear topology on the ring of multivariate power series
+
+- `MvPowerSeries.basis`: the ideals of the ring of multivariate power series
+all coefficients the exponent of which is smaller than some bound vanish.
+
+## Instances :
+
+- `MvPowerSeries.isLinearTopology` : if `R` has a linear topology,
+then the product topology on `MvPowerSeries σ R` is a linear topology.
+This applies in particular when `R` has a discrete topology.
+
+-/
+
+namespace MvPowerSeries
+
+namespace LinearTopology
+
+open scoped Topology
+
+open Set SetLike
+
+/-- The underlying family for the basis of ideals in a multivariate power series ring. -/
+def basis (σ : Type*) (R : Type*) [Ring R] (Jd : TwoSidedIdeal R × (σ →₀ ℕ)) :
```

Comment:
Is it possible to give the definition in term mode? (Of course the proofs can be in tactic mode).